core/src/main/java/com/sequenceiq/cloudbreak/service/parcel/ParcelInfoProvider.java

```diff
@@ -0,0 +1,43 @@
+package com.sequenceiq.cloudbreak.service.parcel;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.cloud.model.ClouderaManagerProduct;
+import com.sequenceiq.cloudbreak.core.CloudbreakImageCatalogException;
+import com.sequenceiq.cloudbreak.core.CloudbreakImageNotFoundException;
+import com.sequenceiq.cloudbreak.service.CloudbreakRuntimeException;
+import com.sequenceiq.cloudbreak.service.image.ImageService;
+import com.sequenceiq.cloudbreak.service.image.StatedImage;
+
+@Component
+public class ParcelInfoProvider {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ParcelInfoProvider.class);
+
+    @Inject
+    private ImageService imageService;
+
+    @Inject
+    private ClouderaManagerProductTransformer clouderaManagerProductTransformer;
+
+    public Set<String> getPreWarmParcelNamesFromImage(Long stackId) {
+        try {
+            StatedImage currentImage = imageService.getCurrentImage(stackId);
+            return clouderaManagerProductTransformer.transform(currentImage.getImage(), false, true)
```

Comment:
`com.sequenceiq.cloudbreak.service.upgrade.sync.component.ImageReaderService#getParcels` has the same logic implemented, pls re-use that, maybe the whole logic can be in that service, too.